File: runtime/python/units/rt/f32_demote.py
# requires: rt/f32, rt/f32_bits, rt/f32_from_bits
@staticmethod
def f32_demote(x):
    if x != x:
        return Rt.f32_from_bits((Rt.f32_bits(x) & 0x80000000) | 0x7FC00000)
    return Rt.f32(x)

File: runtime/go/units/rt/f32_demote.go
func (rt) f32_demote(x float64) float32 {
    if math.IsNaN(x) {
        b := math.Float64bits(x)
        return math.Float32frombits(uint32(b>>32)&0x80000000 | 0x7fc00000)
    }
    return float32(x)
}
